pyhf/tensor/mxnet_backend.py

```diff
@@ -165,19 +179,85 @@ def exp(self, tensor_in):
         return nd.exp(tensor_in)
 
     def stack(self, sequence, axis=0):
-        pass
+        """
+        Join a sequence of arrays along a new axis.
+
+        The axis parameter specifies the index of the new axis in the dimensions
+        of the result. For example, if axis=0 it will be the first dimension and
+        if axis=-1 it will be the last dimension.
+
+        Args:
+            sequence: sequence of arrays
+            axis: the axis along which to join the arrays
+
+        Returns:
+            MXNet NDArray: ndarray comprised of the elements of the sequence
+        """
+        return nd.stack(*sequence, axis=axis)
 
     def where(self, mask, tensor_in_1, tensor_in_2):
-        pass
+        """
+        Apply a boolean selection mask to the elements of the input tensors
+
+        Example:
+            >>> mxnet_backend.where(
+                mxnet_backend.astensor([1, 0, 1]),
+                mxnet_backend.astensor([1, 1, 1]),
+                mxnet_backend.astensor([2, 2, 2]))
+            >>> [1. 2. 1.]
+
+        Args:
+            mask: Boolean mask (boolean or tensor object of booleans)
+            tensor_in_1: tensor object
+            tensor_in_2: tensor object
+
+        Returns:
+            MXNet NDArray: The result of the mask being applied to the tensors
+        """
+        mask = self.astensor(mask)
+        tensor_in_1 = self.astensor(tensor_in_1)
+        tensor_in_2 = self.astensor(tensor_in_2)
+        return nd.multiply(mask, tensor_in_1) + nd.multiply(nd.subtract(1, mask), tensor_in_2)
 
     def concatenate(self, sequence):
-        pass
+        """
+        Join the elements of the sequence
+
+        Args:
+            sequence: the sequence of arrays to join
+
+        Returns:
+            MXNet NDArray: The ndarray of the joined elements
+        """
+        return nd.concat(*sequence, dim=0)
 
     def simple_broadcast(self, *args):
-        pass
+        """
+        Does this work?
+        There should be a more MXNet-style way to do this
+        """
+        broadcast = []
+        max_dim = max(map(len, args))
+        for arg in args:
+            broadcast.append(self.astensor(arg)
+                             if len(arg) > 1 else arg * self.ones(max_dim))
+        return broadcast
 
     def poisson(self, n, lam):
-        pass
+        return self.normal(n, lam, self.sqrt(lam))
 
     def normal(self, x, mu, sigma):
```

Comment:
You can probably use these APIs

https://mxnet.incubator.apache.org/api/python/symbol/symbol.html#mxnet.symbol.random.normal

Reply:
but for poisson, let's stick for now to a gaussian approximation with mu = lambda , sigma = sqrt(lambda) since we need continuous poissons

Reply:
Hm, maybe I'm misunderstanding what you mean, but we want the pdf of the distribution evaluated at `x`, yes? The MXNet distribution generators return sampling of the distributions given the parameters you pass, so [`mx.nd.random.normal()`](https://mxnet.incubator.apache.org/api/python/ndarray/random.html#mxnet.ndarray.random.normal) is akin to [`np.random.normal()`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.normal.html). I might have missed another API though.

Yeah, I follow you RE: the Poissons.

Reply:
ah right, i didn't look to closely at the api seems like it more close to np.random stan scipy.stats.. sorry about that!

Reply:
No worries! Just wanted to make sure I wasn't being dumb. :)